### ISSUE TITLE
semi rule reports beginning of next line on VariableDeclarations missing a semicolon

### ISSUE THREAD

**Issue by iancmyers** [eslint/eslint#670]
The `semi` rule reports the beginning of the line after the `VariableDeclaration` with a missing semicolon.

``` js
var foo = "bar"

// Expected: 1:15 Missing semicolon.
// Actual: 2:0 Missing semicolon.
```

This happens because Esprima reports the `VariableDeclaration` node's location differently based on whether or not a semicolon is present. If a semicolon is present, the end location of the node is the location of the semicolon. If a semicolon is not present, the end location is the beginning of the line after the `VariableDeclaration`.

Perhaps we should report column just after the node's last token instead of the end of the node? Ideally, I should just be able to type `;` at the reported location to fix this issue.
